app/models/miq_event_definition.rb

```diff
@@ -64,7 +64,7 @@ def self.import_from_hash(event, options = {})
 
   def etype
     set = memberof.first
-    raise "unexpected error, no type found for event #{name}" if set.nil?
+    _log.error("No type found for event #{name}") if set.nil?
     set
```

Comment:
@gmcculloug For the first reference, the handling of nil value of event.etype has been added.

For the second reference, it is called for an existing alert. The alert is created only when the based on event has valid event set. So the existing alert should have a valid event.etype.  

Policy controller refers to event.etype as well. But that is for events with valid event set. 

So in all these cases, only when adding a new alert, we need to check for the nil event.etype and skip those events if they don't have a valid event set.  So the alerts can be created only based on valid events.